src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -123,6 +138,29 @@ public function createFilterPaginator($criteria = array(), $sorting = array(), $
     }
 
     /**
+     * Get the product data for the stock page.
+     *
+     * @param integer $id
+     *
+     * @return null|ProductInterface
+     */
+    public function findForStockPage($id)
```

Comment:
Repository should not know about pages, maybe better name is `findStockJoined()`?